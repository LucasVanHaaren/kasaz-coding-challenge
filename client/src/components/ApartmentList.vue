<template>
  <div>
    <div v-if="loading" class="text-center loader">
      <v-progress-circular indeterminate color="green" />
    </div>
    <div v-else>
      <template v-for="(apartment, index) in apartments">
        <Apartment v-bind="apartment" :key="index" class="apartment" />
      </template>
    </div>
    <div class="text-center loader" v-if="!loading && !apartments.length">
      <h1>Your search seems to have no results</h1>
    </div>
    <!-- <div v-else>
      <v-container v-if="$vuetify.breakpoint.mdAndUp">
        <v-row>
          <v-col>
            <template v-for="(apartment, index) in apartments">
              <Apartment v-bind="apartment" :key="index" class="apartment" />
            </template>
          </v-col>
          <v-col>
            <h1>hey</h1>
          </v-col>
        </v-row>
      </v-container>
      <template v-else v-for="(apartment, index) in apartments">
        <Apartment v-bind="apartment" :key="index" class="apartment" />
      </template>
    </div>-->
  </div>
</template>

<script>
import Apartment from '@/components/Apartment.vue';

export default {
  name: 'ApartmentList',
  components: {
      Apartment,
  },
  props: {
      loading: {
          type: Boolean,
          required: true
      },
      apartments: {
          type: Array,
          required: true
      }
  }
};
</script>

<style lang="scss" scoped>
.apartment {
  margin-top: 0.5em;
}

.loader {
  margin-top: 3em;
}
</style>