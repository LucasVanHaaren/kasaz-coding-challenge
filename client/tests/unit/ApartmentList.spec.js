import { shallowMount, mount } from '@vue/test-utils'
import ApartmentList from '@/components/ApartmentList.vue'

describe('ApartmentList.vue', () => {

  const apartments = [
    {
      title: "Title 1",
      price: 150000,
      sqm: 110,
      bedroomsCount: 2,
      bathroomsCount: 1
    },
    {
      title: "Title 2",
      price: 250000,
      sqm: 120,
      bedroomsCount: 4,
      bathroomsCount: 2
    }
  ];

  it('match snapshot', () => {
    const wrapper = shallowMount(ApartmentList, {
      propsData: {
        loading: false,
        apartments: apartments
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders data correctly', () => {
    const wrapper = mount(ApartmentList, {
      propsData: {
        loading: false,
        apartments: apartments
      }
    });
    expect(wrapper.findAll('.apartment').length).toBe(2);
  });
  
  it('renders loader correctly', () => {
    const wrapper = shallowMount(ApartmentList, {
      propsData: {
        loading: true,
        apartments: apartments
      }
    });
    expect(wrapper.find('.loader').isVisible()).toBe(true);
  });
  
  it('renders no-data msg correctly', () => {
    const wrapper = shallowMount(ApartmentList, {
      propsData: {
        loading: false,
        apartments: []
      }
    });
    expect(wrapper.find('h1').isVisible()).toBe(true);
    expect(wrapper.find('h1').text()).toBe("Your search seems to have no results");
  });
})
