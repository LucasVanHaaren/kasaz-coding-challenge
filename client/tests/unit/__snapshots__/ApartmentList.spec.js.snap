// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApartmentList.vue match snapshot 1`] = `
<div>
  <div>
    <apartment-stub
      bathroomscount="1"
      bedroomscount="2"
      class="apartment"
      price="150000"
      sqm="110"
      thumbnail="/pictures/default.jpg"
      title="Title 1"
    />
    <apartment-stub
      bathroomscount="2"
      bedroomscount="4"
      class="apartment"
      price="250000"
      sqm="120"
      thumbnail="/pictures/default.jpg"
      title="Title 2"
    />
  </div>
   
  <!---->
</div>
`;
